lib/src/facts/freebsd/processor_resolver.cc

```diff
@@ -0,0 +1,50 @@
+#include <internal/facts/freebsd/processor_resolver.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <sys/types.h>
+#include <sys/sysctl.h>
+
+using namespace std;
+
+namespace facter { namespace facts { namespace freebsd {
+
+    processor_resolver::data processor_resolver::collect_data(collection& facts)
+    {
+        auto result = posix::processor_resolver::collect_data(facts);
+        size_t len;
+        int mib[2];
+        mib[0] = CTL_HW;
+
+        // Get the logical count of processors
+        len = sizeof(result.logical_count);
+        mib[1] = HW_NCPU;
+
+        if (sysctl(mib, 2, &result.logical_count, &len, nullptr, 0) == -1) {
+            LOG_DEBUG("sysctl hw.ncpu failed: %1% (%2%): logical processor count is unavailable.", strerror(errno), errno);
+        }
+
+        // For each logical processor, collect the model name
+        if (result.logical_count > 0) {
+            // Note: we're using the same description string for all logical processors:
+            // using different CPUs is not even likely to work.
+            vector<char> buffer(256);
+
+            while (true) {
+                size_t size = buffer.size();
+                mib[1] = HW_MODEL;
+                if (sysctl(mib, 2, buffer.data(), &size, nullptr, 0) == 0) {
+                    buffer.resize(size + 1);
+                    result.models.resize(result.logical_count, buffer.data());
+                    break;
+                }
+                if (errno != ENOMEM) {
+                    LOG_DEBUG("sysctl hw.model failed: %1% (%2%): processor models are unavailable.", strerror(errno), errno);
+                    break;
+                }
+                buffer.resize(buffer.size() * 2);
+            }
+        }
```

Comment:
Yeah, I hit my head against the wall with this for a while. It looks like FreeBSD's [kenv](https://www.freebsd.org/cgi/man.cgi?kenv(1)) method seems like the way to go:

```c
#ifdef ENABLE_LLDPMED
	/* Fill in inventory stuff:
	    - hardware version: smbios.system.version
	    - firmware version: smbios.bios.version
	    - software version: `uname -r`
	    - serial number: smbios.system.serial
	    - manufacturer: smbios.system.maker
	    - model: smbios.system.product
	    - asset: smbios.chassis.tag
	*/

static char*
dmi_get(char *file)
{
	char buffer[100] = {};

	log_debug("localchassis", "DMI request for %s", file);
	if (kenv(KENV_GET, file, buffer, sizeof(buffer) - 1) == -1) {
		log_debug("localchassis", "cannot get %s", file);
		return NULL;
	}
	if (strlen(buffer))
		return strdup(buffer);
	return NULL;
}

char*
dmi_sn()
{
	return dmi_get("smbios.system.serial");
}
```